apps/performance-tests/src/util/Datasets.ts

```diff
@@ -127,4 +164,203 @@ export class Datasets {
+            modelId: drawingModelId,
+            categoryId: drawingCategoryId,
+          }).id;
+        }
+      }
+    });
+
+    console.log(`${numElements} elements: Done.`);
+  }
+
+  private static async createNestedSubjectModel(name: string, localPath: string, numElements: number) {
+    console.log(`${numElements} elements: Creating...`);
+
+    await createIModel(name, localPath, async (builder) => {
+      const numberOfGroups = 20;
+      const elementsPerGroup = numElements / numberOfGroups;
+
+      for (let i = 0; i < numberOfGroups; ++i) {
+        let parentId: string | undefined;
+
+        for (let j = 0; j < elementsPerGroup; ++j) {
+          parentId = insertSubject({
+            parentId,
+            builder,
+            codeValue: `subject_${i}_${j}`,
+          }).id;
+        }
+
+        insertPhysicalModelWithPartition({ builder, codeValue: `model_${i}`, partitionParentId: parentId });
+      }
+    });
+
+    console.log(`${numElements} elements: Done.`);
+  }
+
+  private static async createNestedCategoriesModel(name: string, localPath: string, numElements: number) {
+    console.log(`${numElements} elements: Creating...`);
+    await createIModel(name, localPath, async (builder) => {
+      const { id: categoryId } = insertSpatialCategory({ builder, fullClassNameSeparator: ":", codeValue: "My Category" });
+
+      for (let i = 0; i < numElements; ++i) {
+        insertSubCategory({
+          parentCategoryId: categoryId,
+          builder,
+          codeValue: `${i}`,
+        });
+      }
+    });
+
+    console.log(`${numElements} elements: Done.`);
+  }
+
+  private static async createNestedGroupModel(name: string, localPath: string, numElements: number) {
+    console.log(`${numElements} elements: Creating...`);
+
+    const numGroups = 10;
+    const membersPerGroup = 100;
+    const elementsPerMember = numElements / (numGroups * membersPerGroup);
+
+    await createIModel(name, localPath, async (builder) => {
+      const { id: groupModelId } = insertGroupInformationModelWithPartition({ builder, codeValue: "group information model" });
+      const { id: modelId } = insertPhysicalModelWithPartition({ builder, codeValue: "test physical model" });
+      const { id: categoryId } = insertSpatialCategory({ builder, fullClassNameSeparator: ":", codeValue: "My Category" });
+
+      for (let groupIdx = 0; groupIdx < numGroups; ++groupIdx) {
+        const { id: groupId } = insertGroupInformationElement({ builder, modelId: groupModelId });
+
+        for (let i = 0; i < membersPerGroup; ++i) {
+          let parentId: string | undefined;
+
+          for (let j = 0; j < elementsPerMember; ++j) {
+            parentId = insertPhysicalElement({ builder, parentId, modelId, categoryId }).id;
+            if (j === 0) {
+              builder.insertRelationship({ sourceId: groupId, targetId: parentId, classFullName: "BisCore.ElementGroupsMembers" });
+            }
+          }
+        }
```

Comment:
Do we need the grouped elements have children? Maybe it would be better to have more groups with more direct members, without any child elements - that would isolate the grouping case from elements' hierarchy case.

Reply:
I added the children as the hilite elements query contains a step for traversing children of group members which would otherwise not be tested. Should I still remove the children in this case?

Reply:
Okay, I guess that makes sense. But maybe we should focus more on grouped elements here, rather than child ones, and increase `membersPerGroup` to `1000`?

Reply:
And I think the function description needs to be clarified. I understand a "group element" as a `GroupInformationElement`, but this kind of element doesn't have a category and doesn't belong to physical model. Instead, we got 10 group (`GroupInformationElement`) elements and each of them have related physical elements through the `ElementGroupsMembers` relationship.

Reply:
Increased the number of group members and fixed the description.